projects/pastanaga/src/lib/controls/control.model.ts

```diff
@@ -1,8 +1,30 @@
+export interface IControlModel {
+    id: string;
+    label: string;
+    subLabel?: string;
+    labelIcons?: LabelIcon[];
+    value?: string;
+    help?: string;
+    icon?: string;
+    isSelected?: boolean;
+    isDisabled?: boolean;
+    isFiltered?: boolean;
+    isIndeterminate?: boolean;
+    isExpanded?: boolean;
+    isHidden?: boolean;
+    isPrivate?: boolean;
+
+    children?: ControlModel[];
```

Comment:
An interface containing a derived Class? I think this should be  `IControlModel[];`

Reply:
disagree. I don't see why an interface should not type a property with only primitive types or interfaces. Here I expect `children` to be of type `ControlModel`, not just to respect `IControlModel` interface.

Reply:
I mean, it's a bit weird for me because you need a `IControlModel` in the constructor to create a `ControlModel` object so, if the `IControlModel` object has actually several `ControlModel` objects in its children property its like... the chicken or the egg dilemma? :thinking:

Reply:
`IControlModel` is not an object, it's an interface.
`ControlModel` is an object, a class, implementing this interface.
`ControlModel` has a constructor, which takes `IControlModel` as data parameter. 
Inside the constructor, we do not instantiate anything: we don't do `new ControlModel(...)` in `ControlModel` constructor.
So you don't have any chicken/egg dilemna here. 

When you want to instantiate a ControlModel which have children, you'll do something like
```typescript
        new ControlModel({id: 'africa', label: 'Africa', children: [
            new ControlModel({id: 'morocco', label: 'Morocco'}),
            new ControlModel({id: 'tunisia', label: 'Tunisia'}),
            new ControlModel({id: 'senegal', label: 'Senegal'}),
        ]})
```

Reply:
That's why I wanted to propose you to just let the constructor handle the creation of new ControlModels, with that change I think it would be easier to use:
```
new ControlModel({id: 'africa', label: 'Africa', children: [
            {id: 'morocco', label: 'Morocco'},
            {id: 'tunisia', label: 'Tunisia'},
            {id: 'senegal', label: 'Senegal'},
]})
```

Reply:
@LuisPalomo I think it is just a semantic issue here. `IControlModel` is not really an interface (as `ControlModel` is not implementing this interface), it is just a way to type the payload of the `ControlModel` constructor, and we do that technically by creating an interface.
If we named it `ControlModelData` rather than `IControlModel`, it would be less misleading maybe.
It is totally fine to have `children?: ControlModel[]` in there, just as fine as if we passed all the params to the constructor directly, including the children:
```
constructor(id: string, label: string, children: ControlModel[], …)
```
`IControlModel` just allows to manage this long bunch of parameters into a proper way.

And passing existing `ControlModel` as children is more flexible than creating them inside the constructor as you can do something like:
```
myControl1: ControlModel;
myControl2: ControlModel;
myControlGroup = new ControlModel({id: 'group', children: [this.myControl1, this.myControl2]}); 
```
so you keep the ability to instantiate and manipulate the children from the outside.